### PR TITLE
Add cmake test fixes to go with .mod file changes made recently

### PR DESCRIPTION
**Pull request by seando-adsk**
Add cmake test fixes to go with .mod file changes made here:
https://github.com/Autodesk/maya-usd/pull/2867

### CODE REVIEW

File: cmake/test.cmake [hunk 1]
```diff
@@ -43,6 +43,7 @@ endfunction()
 #                        that mayaUsd_add_test manages (and will append) are:
 #                            PATH
 #                            PYTHONPATH
+#                            PXR_USD_WINDOWS_DLL_PATH
```
Comment (seando-adsk) [inline cmake/test.cmake:46]: In PR #2867 a client added this env var to the .mod files. I've updated our testing environment to match.

File: cmake/test.cmake [hunk 2]
```diff
@@ -221,64 +225,87 @@ finally:
...
-         "${CMAKE_INSTALL_PREFIX}/lib/scripts")
-    if (IS_LINUX)
-        # On Linux the paths in XBMLANGPATH need a /%B at the end.
-        list(APPEND MAYAUSD_VARNAME_XBMLANGPATH
-             "${CMAKE_INSTALL_PREFIX}/lib/icons/%B")
-    else()
-        list(APPEND MAYAUSD_VARNAME_XBMLANGPATH
-             "${CMAKE_INSTALL_PREFIX}/lib/icons")
+    # MayaUsd core library
+    if (BUILD_MAYAUSD_LIBRARY)
```
Comment (seando-adsk) [inline cmake/test.cmake:229]: Moved the various env vars inside of if blocks based on their respective build option.

File: cmake/test.cmake [hunk 3]
```diff
@@ -221,64 +225,87 @@ finally:
...
+        list(APPEND MAYAUSD_VARNAME_PATH
+             "${CMAKE_INSTALL_PREFIX}/lib")
+        if(IS_WINDOWS)
+            list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH
+                "${CMAKE_INSTALL_PREFIX}/lib")
+        endif()
```
Comment (seando-adsk) [inline cmake/test.cmake:235]: This is new.

File: cmake/test.cmake [hunk 4]
```diff
@@ -221,64 +225,87 @@ finally:
...
+        else()
+            list(APPEND MAYAUSD_VARNAME_XBMLANGPATH
+                 "${CMAKE_INSTALL_PREFIX}/lib/icons")
+        endif()
+        list(APPEND MAYAUSD_VARNAME_PYTHONPATH
+             "${CMAKE_INSTALL_PREFIX}/lib/python")
+        list(APPEND MAYAUSD_VARNAME_${PXR_OVERRIDE_PLUGINPATH_NAME}
+             "${CMAKE_INSTALL_PREFIX}/lib/usd")
+        list(APPEND MAYAUSD_VARNAME_PXR_MTLX_STDLIB_SEARCH_PATHS
+             "${CMAKE_INSTALL_PREFIX}/libraries")
```
Comment (seando-adsk) [inline cmake/test.cmake:253]: Moved up from below.

File: cmake/test.cmake [hunk 5]
```diff
@@ -221,64 +225,87 @@ finally:
...
+    # pxr plugin
+    if (BUILD_PXR_PLUGIN)
+        list(APPEND MAYAUSD_VARNAME_PYTHONPATH
+             "${CMAKE_INSTALL_PREFIX}/plugin/pxr/lib/python")
+        list(APPEND MAYAUSD_VARNAME_PATH
+             "${CMAKE_INSTALL_PREFIX}/plugin/pxr/maya/lib")
+        if(IS_WINDOWS)
+            list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH
+                "${CMAKE_INSTALL_PREFIX}/plugin/pxr/maya/lib")
+        endif()
```
Comment (seando-adsk) [inline cmake/test.cmake:275]: This is new.

File: cmake/test.cmake [hunk 6]
```diff
@@ -221,64 +225,87 @@ finally:
...
+             "${CMAKE_INSTALL_PREFIX}/plugin/al/plugin")
+    endif()
 
     if(IS_WINDOWS AND DEFINED ENV{PYTHONHOME})
         # If the environment contains a PYTHONHOME, also set the path to
         # that folder so that we can find the python DLLs.
         list(APPEND MAYAUSD_VARNAME_PATH $ENV{PYTHONHOME})
+        if(IS_WINDOWS)
+            list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH $ENV{PXR_USD_WINDOWS_DLL_PATH})
+        endif()
```
Comment (seando-adsk) [inline cmake/test.cmake:308]: This is new.

File: cmake/test.cmake [hunk 7]
```diff
@@ -303,7 +332,18 @@ finally:
...
             "${USD_INSTALL_LOCATION}/lib")
+        list(APPEND MAYAUSD_VARNAME_PATH
+            "${USD_INSTALL_LOCATION}/plugin/usd")
+
+        list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH
+            "${USD_INSTALL_LOCATION}/bin")
+        list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH
+            "${USD_INSTALL_LOCATION}/lib")
+        list(APPEND MAYAUSD_VARNAME_PXR_USD_WINDOWS_DLL_PATH
+            "${USD_INSTALL_LOCATION}/plugin/usd")
```
Comment (seando-adsk) [inline cmake/test.cmake:343]: This is new.

File: cmake/test.cmake [hunk 8]
```diff
@@ -303,7 +332,18 @@ finally:
...
     endif()
+    list(APPEND MAYAUSD_VARNAME_PXR_MTLX_STDLIB_SEARCH_PATHS
+            "${PXR_USD_LOCATION}/libraries")
```
Comment (seando-adsk) [inline cmake/test.cmake:346]: Moved here (into USD section) from above

**Review by ppt-adsk — APPROVED** [review]

**Comment by seando-adsk** [comment]
Sorry I accidentally added PierreT as reviewer when I wanted to add PierreB.

**Review by pierrebai-adsk — APPROVED** [review]
Not expert enough to debug cmake visually, but did not see any abvious problem. Looks good to me.

**Pull request merged**